tests/base.py

```diff
@@ -5,37 +5,12 @@
 
 from syncer import sync
 
-from pyppeteer import launch
+from pyppeteer import Pyppeteer
+from pyppeteer.browser import Browser
 from pyppeteer.util import get_free_port
 
 from .server import get_application
 
 DEFAULT_OPTIONS = {'args': ['--no-sandbox']}
 
-
-class BaseTestCase(unittest.TestCase):
-    @classmethod
-    def setUpClass(cls):
-        cls.port = get_free_port()
-        cls.app = get_application()
-        cls.server = cls.app.listen(cls.port)
-        cls.browser = sync(launch(DEFAULT_OPTIONS))
-        cls.url = 'http://localhost:{}/'.format(cls.port)
-
-    @classmethod
-    def tearDownClass(cls):
-        sync(cls.browser.close())
-        cls.server.stop()
-
-    def setUp(self):
-        self.context = sync(self.browser.createIncognitoBrowserContext())
-        self.page = sync(self.context.newPage())
-        self.result = False
-
-    def tearDown(self):
-        sync(self.context.close())
-        self.context = None
-        self.page = None
-
-    def set_result(self, value):
-        self.result = value
+browser: Browser = sync(Pyppeteer().launch(args='--no-sandbox'))
```

Comment:
`--no-sandbox` seems like it might effect some tests. I'm not sure about changings this as sandbox is a bit problematic to setup but we definitely should document this somewhere like:  
```suggestion
# NOTE: test browser is using --no-sandbox flag might cause inconsistencies 
browser: Browser = sync(Pyppeteer().launch(args='--no-sandbox'))
```

Reply:
I don't _think_ that should be the case, but I don't know enough about it to say it's not impossible interferes. And if it did cause the bug, I'm pretty sure it would be a Chrome bug. But I could easily set it up to use an env var for that.

Here's what they say in the puppeteer README:

> in order to protect the host environment from untrusted web content, Chrome uses [multiple layers of sandboxing](https://chromium.googlesource.com/chromium/src/+/HEAD/docs/linux_sandboxing.md). For this to work properly, the host should be configured first. If there's no good sandbox for Chrome to use, it will crash with the error `No usable sandbox!`.
> If you **absolutely trust** the content you open in Chrome, you can launch Chrome with the `--no-sandbox` argument:
```js
const browser = await puppeteer.launch({args: ['--no-sandbox', '--disable-setuid-sandbox']});
```
**NOTE**: Running without a sandbox is **strongly discouraged**. Consider configuring a sandbox instead.


OTOH, it is interesting to note that the dockerfile for puppeteer testing includes a workaround for not using it: 
 
> # Add user so we don't need --no-sandbox.

But doesn't say why.

Reply:
I guess we can just keep this in mind for the time being.